File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Controllers/PermissaoController.cs
using JsonWebToken.Code;
using JsonWebToken.Model;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using RestSharp;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Configuration;
using System.Web.Http;

namespace JsonWebToken.Controllers
{

  [Authorize]
  public class PermissaoController : ApiController
  {

    [HttpPost]
    public IHttpActionResult PostAuthenticateAD(string user, string password)
    {
      try
      {
        var client = new RestClient(WebConfigurationManager.AppSettings["urlAuthenticationAPI"]);

        var request = new RestRequest(string.Format("/Data/authenticateAD?user={0}&password={1}", user, password), Method.POST)
        {
          RequestFormat = DataFormat.Json
        };

        var response = client.Execute(request);

        if (response.StatusCode == HttpStatusCode.OK)
        {
          AuthenticateAdModel resp = JsonConvert.DeserializeObject<AuthenticateAdModel>(response.Content);

          return Ok(resp);
        }
        else
        {
          return Unauthorized();
        }
      }
      catch (Exception ex)
      {
        return BadRequest(ex.Message);
      }
    }
  }
}

File: Backend/Administracao.ConciliacaoAPI/Controllers/PermissaoController.vb
Imports JsonWebToken.Model
Imports Administracao.ConciliacaoAPI.Model
Imports Newtonsoft.Json
Imports RestSharp
Imports System
Imports System.Net
Imports System.Web.Configuration
Imports System.Web.Http

Namespace Controllers

    <Authorize>
    Public Class PermissaoController
        Inherits ApiController

        <HttpPost>
        Public Function PostAuthenticateAD(user As String, password As String) As AuthenticateAdModel
            Try
                Dim client = New RestClient(WebConfigurationManager.AppSettings("urlAuthenticationAPI"))

                Dim request = New RestRequest(String.Format("/Data/authenticateAD?user={0}&password={1}", user, password), Method.POST)
                request.RequestFormat = DataFormat.Json

                Dim response = client.Execute(request)

                If (response.StatusCode = HttpStatusCode.OK) Then
                    Return JsonConvert.DeserializeObject(Of AuthenticateAdModel)(response.Content)
                Else
                    Return New AuthenticateAdModel()
                End If

            Catch ex As Exception
                Throw ex
            End Try
        End Function
    End Class
End Namespace
